src/foam/u2/view/ArrayView.js

```diff
@@ -128,7 +153,12 @@ foam.CLASS({
       this.addClass();
 
       this
-        .add(this.slot(function(data2_, valueView) {
+        .add(this.slot(function(data2_, valueView, arrayLength_) {
+          if ( ! self.allowDuplicates) {
+            // Disable adding if no duplicates and all uniques already assigned
+            this.enableAdding = self.data.length !== self.arrayLength_;
+          }
```

Comment:
This hides the add button if all labels are assigned
Not sure if this is desired or we should still show Add button but disabled?

Reply:
I would say we should disable it instead by just setting the `isEnabled` on the `addRow` action

Reply:
I have made the recommended change